examples/showVisitSkyMap.py

```diff
@@ -71,6 +72,16 @@ def main(rootDir, tract, visits, ccds=None, ccdKey='ccd', showPatch=False, saveF
                     decs += dec
                     color = ('r', 'b', 'c', 'g', 'm')[i_v%5]
                     pyplot.fill(ra, dec, fill=True, alpha=0.2, color=color, edgecolor=color)
+                    if showCcds:
+                        minPoint = afwGeom.Point2D(min(ra), min(dec))
+                        maxPoint = afwGeom.Point2D(max(ra), max(dec))
+                        currentBbox = afwGeom.Box2D(minPoint, maxPoint)
```

Comment:
The earlier bbox is a Box2I, which has integer values for the corners.  The
right ascension and declination are then returned from the function
bboxToRaDec.  It seems when the program runs, the bbox loaded from each ccd
have the same integer values for their corners, making it impossible to
check overlaps with the bbox instances.  The currentBbox is a Box2D (with
double values for the corners), which are constructed using the extracted
Ra and Dec values from the bbox, which we can then use to check overlaps.

Maybe the currentBbox should be renamed to make this less obscure?

On Thu, Jul 14, 2016 at 11:36 AM, hchiang2 <notifications@github.com> wrote:

> In examples/showVisitSkyMap.py
> <https://github.com/lsst/skymap/pull/2#discussion_r70838680>:
>
> > @@ -71,6 +72,16 @@ def main(rootDir, tract, visits, ccds=None, ccdKey='ccd', showPatch=False, saveF
> >                      decs += dec
> >                      color = ('r', 'b', 'c', 'g', 'm')[i_v%5]
> >                      pyplot.fill(ra, dec, fill=True, alpha=0.2, color=color, edgecolor=color)
> > +                    if showCcds:
> > +                        minPoint = afwGeom.Point2D(min(ra), min(dec))
> > +                        maxPoint = afwGeom.Point2D(max(ra), max(dec))
> > +                        currentBbox = afwGeom.Box2D(minPoint, maxPoint)
>
> Is this currentBbox the same as the bbox a few lines earlier? If so,
> maybe no need to create a new one?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lsst/skymap/pull/2/files/cc135b780f85a91e62a67c306157966969dc51ce#r70838680>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ATEgY4Kt7f9DGQXYKLkHulYip6MUkjAaks5qVmWJgaJpZM4JLgRS>
> .
>



-- 
Sam Piehl